src/System.Text.Utf8String/System/Text/Utf8Span.cs

```diff
@@ -0,0 +1,782 @@
+                // Iterate over prefix set
+                while (itPrefix.MoveNext())
+                {
+                    if (it.Current == itPrefix.Current)
+                    {
+                        // Character found, don't check further
+                        found = true;
+                        break;
+                    }
+                }
+
+                if (!found)
+                {
+                    // Reached the end, char was not found
+                    break;
+                }
+
+                itPrefix.Reset();
+            }
+
+            return Substring(it.PositionInCodeUnits);
+        }
+
+        public Utf8Span TrimEnd()
+        {
+            CodePointReverseEnumerator it = CodePoints.GetReverseEnumerator();
+            while (it.MoveNext() && Utf8Helper.IsWhitespace(it.Current))
+            {
+            }
+
+            return Substring(0, it.PositionInCodeUnits);
+        }
+
+        public Utf8Span TrimEnd(uint[] trimCodePoints)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Utf8Span TrimEnd(byte[] trimCodeUnits)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Utf8Span TrimEnd(Utf8Span trimCharacters)
+        {
+            if (trimCharacters == Empty)
+            {
+                // Trim Whitespace
+                return TrimEnd();
+            }
+
+            CodePointReverseEnumerator it = CodePoints.GetReverseEnumerator();
+            CodePointEnumerator itPrefix = trimCharacters.GetCodePointEnumerator();
+
+            while (it.MoveNext())
+            {
+                bool found = false;
+                // Iterate over prefix set
+                while (itPrefix.MoveNext())
+                {
+                    if (it.Current == itPrefix.Current)
+                    {
+                        // Character found, don't check further
+                        found = true;
+                        break;
+                    }
+                }
+
+                if (!found)
+                {
+                    // Reached the end, char was not found
+                    break;
+                }
+
+                itPrefix.Reset();
+            }
+
+            return Substring(0, it.PositionInCodeUnits);
+        }
+
+        public Utf8Span Trim()
+        {
+            return TrimStart().TrimEnd();
+        }
+
+        public Utf8Span Trim(uint[] trimCodePoints)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Utf8Span Trim(byte[] trimCodeUnits)
+        {
+            throw new NotImplementedException();
+        }
+
+        // TODO: Name TBD, CopyArray? GetBytes?
+        public byte[] CopyBytes()
+        {
+            return _buffer.ToArray();
+        }
+
+        public byte[] CopyCodeUnits()
+        {
+            throw new NotImplementedException();
+        }
+
+        public static bool IsWhiteSpace(byte codePoint)
```

Comment:
This should not be public. If we have code point APIs they should be in general Unicode type and should be 4 byte int.